memory/title_sequence_manager.py

```diff
@@ -28,36 +28,41 @@ def __init__(self):
 
     def update(self):
         try:
-            self.memory.update()
-            if self.memory.ready_for_updates():
-                singleton_ptr = self.memory.get_singleton_by_class_name(
-                    "TitleSequenceManager"
-                )
-
-                self.base = self.memory.get_class_base(singleton_ptr)
-                self.fields_base = self.memory.get_class_fields_base(singleton_ptr)
-                self.title_screen = self.memory.get_field(
-                    self.fields_base, "titleScreen"
-                )
-
-                # Update fields
-                self.title_position_set = False
-                self._read_load_save_done()
-                self._read_pressed_start()
-                self._read_continue_selected()
-                self._read_new_game_selected()
-                self._read_load_game_selected()
-                self._read_options_selected()
-                self._read_how_to_play_selected()
-                self._read_quit_selected()
+            if self.memory.ready_for_updates:
+                if self.base is None or self.fields_base is None:
+                    singleton_ptr = self.memory.get_singleton_by_class_name(
+                        "TitleSequenceManager"
+                    )
+
+                    if singleton_ptr is None:
+                        return
+
+                    self.base = self.memory.get_class_base(singleton_ptr)
+
+                    if self.base == 0x0:
+                        return
+
+                    self.fields_base = self.memory.get_class_fields_base(singleton_ptr)
+                    self.title_screen = self.memory.get_field(
+                        self.fields_base, "titleScreen"
+                    )
+                else:
+                    # Update fields
+                    self.title_position_set = False
+                    self._read_load_save_done()
+                    self._read_pressed_start()
+                    self._read_continue_selected()
+                    self._read_new_game_selected()
+                    self._read_load_game_selected()
+                    self._read_options_selected()
+                    self._read_how_to_play_selected()
+                    self._read_quit_selected()
 
                 if not self.title_position_set:
                     self.title_cursor_position = TitleCursorPosition.NONE
-            else:
-                self.__init__()
-        except Exception:
-            self.title_cursor_position = TitleCursorPosition.NONE
-            return
+        except Exception as _e:
+            # print(f"Title Sequence Manager Reloading {type(_e)}")
+            self.__init__()
 
     def _read_title_cursor_position(self):
```

Comment:
This one seems to be different from all the other `_read_x` functions in that it doesn't do anything. Maybe refactor it out and just look at the member variable instead where it's used.

If you leave it, add a type hint for the return value.